src/libtexture/imagecache_pvt.h

```diff
@@ -842,6 +842,14 @@ class ImageCacheImpl final : public ImageCache {
                                 int subimage, int miplevel, ustring dataname,
                                 TypeDesc datatype, void* data);
 
+    virtual bool get_image_info_type(ustring filename, int subimage,
+                                     int miplevel, ustring dataname,
+                                     TypeDesc& datatype);
+    virtual bool get_image_info_type(ImageCacheFile* file,
```

Comment:
Let's just have these return the TypeDesc directly, since that's the way the equivalent works in other places we have a "get attribute type" kind of method. You don't need the bool return since there is already a special TypeDesc value (`TypeUnknown`) to indicate failure to find a matching item.